packages/manager/apps/zimbra/src/pages/dashboard/MailingLists/MailingListSettings.page.tsx

```diff
@@ -213,59 +216,59 @@ export default function MailingListSettings({
     setSelectedDomainOrganization(organizationLabel);
   };
 
-  const handleSuccess = () => {
-    goBack();
-    addSuccess(
-      <OsdsText
-        color={ODS_THEME_COLOR_INTENT.text}
-        size={ODS_THEME_TYPOGRAPHY_SIZE._400}
-        level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
-        hue={ODS_THEME_COLOR_HUE._500}
-      >
-        {t('zimbra_mailinglist_add_success_message')}
-      </OsdsText>,
-      true,
-    );
-  };
-
-  const handleError = ({ response }: ApiError) => {
-    goBack();
-    addError(
-      <OsdsText
-        color={ODS_THEME_COLOR_INTENT.text}
-        size={ODS_THEME_TYPOGRAPHY_SIZE._400}
-        level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
-        hue={ODS_THEME_COLOR_HUE._500}
-      >
-        {t('zimbra_mailinglist_add_error_message', {
-          error: response.data.message,
-        })}
-      </OsdsText>,
-      true,
-    );
-  };
-
-  const handleNewMailingListClick = () => {
-    const dataBody = getDataBody(form);
-
-    postZimbraPlatformMailingList(
-      platformId,
-      dataBody as MailingListBodyParamsType,
-    )
-      .then(handleSuccess)
-      .catch(handleError);
-  };
-
-  const handleModifyMailingListClick = () => {
-    const dataBody = getDataBody(form);
+  const { mutate: addOrEditMailingList, isPending: isSending } = useMutation({
+    mutationFn: (params: MailingListBodyParamsType) => {
+      return editMailingListId
+        ? putZimbraPlatformMailingList(platformId, editMailingListId, params)
+        : postZimbraPlatformMailingList(platformId, params);
+    },
+    onSuccess: () => {
+      addSuccess(
+        <OsdsText
+          color={ODS_THEME_COLOR_INTENT.text}
+          size={ODS_THEME_TYPOGRAPHY_SIZE._100}
+          level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+          hue={ODS_THEME_COLOR_HUE._500}
+        >
+          {t(
+            editMailingListId
+              ? 'zimbra_mailinglist_edit_success_message'
+              : 'zimbra_mailinglist_add_success_message',
+          )}
+        </OsdsText>,
+        true,
+      );
+    },
+    onError: (error: ApiError) => {
+      addError(
+        <OsdsText
+          color={ODS_THEME_COLOR_INTENT.text}
+          size={ODS_THEME_TYPOGRAPHY_SIZE._100}
+          level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+          hue={ODS_THEME_COLOR_HUE._500}
+        >
+          {t(
+            editMailingListId
+              ? 'zimbra_mailinglist_edit_error_message'
```

Comment:
same for constants here

Reply:
same we don't store translations keys as constants